### PR TITLE
img: fix image_name() when image is empty

### PR DESCRIPTION
**Pull request by nviennot**
When an image is opened but errored with a ENOENT error, the image is still valid. Later on, do_pb_read_one() can fail and will invoke image_name(). The image fd is EMPTY_IMG_FD (-404). read_fd_link fails.

### CODE REVIEW

File: criu/protobuf.c [hunk 1]
```diff
@@ -25,7 +25,7 @@ static char *image_name(struct cr_img *img)
 	int fd = img->_x.fd;
 	static char image_path[PATH_MAX];
 
-	if (read_fd_link(fd, image_path, sizeof(image_path)) > 0)
+	if (fd >= 0 && read_fd_link(fd, image_path, sizeof(image_path)) > 0)
```
Comment (Snorch) [inline criu/protobuf.c:28]: Maybe we can do something like:

```
if (lazy_image(img)) {
        return img->path;
} else if (!empty_image(img)) {
        if (read_fd_link(fd, image_path, sizeof(image_path)) > 0)
                return image_path;
}

return NULL;
```

Comment (nviennot) [reply criu/protobuf.c]: Excellent suggestion. I made some modifications. I still want to do a `fd >= 0` test just in case we add new image failures.

**Comment by nviennot** [comment]
Also, it occurs to me that the there's a `static char image_path[PATH_MAX]` in this function. It seems that we are wasting 4K of resident memory for no good reason. Should this be addressed?

**Comment by nviennot** [comment]
image_path is now on the stack. I wish the stack space was only reserved when about to print an error message though.

File: criu/protobuf.c [hunk 2]
```diff
@@ -48,14 +47,15 @@ static char *image_name(struct cr_img *img)
 
 int do_pb_read_one(struct cr_img *img, void **pobj, int type, bool eof)
 {
+	char imgp[PATH_MAX];
```
Comment (avagin) [inline criu/protobuf.c:50]: could you

Comment (nviennot) [reply criu/protobuf.c:50]: This was my first choice, but then it makes some of the pr_err to wrap line. Should I still rename?

Comment (nviennot) [reply criu/protobuf.c:50]: This was my first choice, but then it makes some of the pr_err to wrap line. Should I still rename?

Comment (avagin) [reply criu/protobuf.c:50]: what do you think about this https://github.com/avagin/criu/commit/797f27605bbd7f9ebf0bd15a5c4e23305fac80dc ?

Comment (nviennot) [reply criu/protobuf.c:50]: sold

**Pull request merged**